R/date_standardization_helpers.R

```diff
@@ -0,0 +1,516 @@
+#' Reformat a given date
+#'
+#' @param x A string of interest
+#' @param format A date format
+#' @keywords internal
+#'
+as_date <- function(x, format = c("ymd", "ydm", "dmy", "mdy", "myd", "dym",
+                                  "Ymd", "Ydm", "dmY", "mdY", "mYd", "dYm")) {
+  fmt <- lubridate::guess_formats(x, format)
+  fmt <- unique(fmt)
+  y   <- as.Date(x, format = fmt[[1L]])
+  for (i in seq_along(fmt)[-1L]) {
+    na    <- is.na(y)
+    if (!any(na)) break
+    y[na] <- as.Date(x[na], format = fmt[i])
+  }
+  return(y)
+}
+
+
+#' Check date time frame
+#'
+#' @param first_date A Date object specifying the first valid date.
+#' The default value is fifty years before the `last_date`
```

Comment:
This value and the function where it is used is inherited from Thibault's code.

My guess is that it is arbitrary and could have been any other value.

Any suggestion?

Reply:
```suggestion
#' The arbitrary default value is fifty years before the `last_date`
```

No alternative suggestion really - it might be informative to say it is arbitrary and not for a specific reason.

Reply:
Thanks @chartgerink - I have accounted for this in commit `84ca479`.